zkevm-circuits/src/state_circuit/lexicographic_ordering.rs

```diff
@@ -348,39 +289,33 @@ fn rw_to_be_limbs(row: &Rw) -> Vec<u16> {
         .collect()
 }
 
-fn upper_limb_difference_possible_values<F: Field>(
-    cur: Queries<F>,
-    prev: Queries<F>,
-) -> Vec<Expression<F>> {
-    let mut result = vec![];
-    let mut partial_sum = 0u64.expr();
-    for (cur_limb, prev_limb) in cur.be_limbs()[..15].iter().zip(&prev.be_limbs()[..15]) {
-        partial_sum = partial_sum * (1u64 << 16).expr() + cur_limb.clone() - prev_limb.clone();
-        result.push(partial_sum.clone())
-    }
-    result
-}
-
-fn lower_limb_difference_possible_values<F: Field>(
+fn rlc_limb_differences<F: Field>(
     cur: Queries<F>,
     prev: Queries<F>,
+    powers_of_randomness: [Expression<F>; 31],
 ) -> Vec<Expression<F>> {
     let mut result = vec![];
     let mut partial_sum = 0u64.expr();
-    for (cur_limb, prev_limb) in cur.be_limbs()[15..].iter().zip(&prev.be_limbs()[15..]) {
-        partial_sum = partial_sum * (1u64 << 16).expr() + cur_limb.clone() - prev_limb.clone();
-        result.push(partial_sum.clone())
+    let powers_of_randomness = once(1.expr()).chain(powers_of_randomness.into_iter());
+    for ((cur_limb, prev_limb), powers_of_randomness) in cur
+        .be_limbs()
+        .iter()
+        .zip(&prev.be_limbs())
+        .zip(powers_of_randomness)
+    {
+        result.push(partial_sum.clone());
+        partial_sum = partial_sum + powers_of_randomness * (cur_limb.clone() - prev_limb.clone());
```

Comment:
```suggestion
    for ((cur_limb, prev_limb), power_of_randomness) in cur
        .be_limbs()
        .iter()
        .zip(&prev.be_limbs())
        .zip(powers_of_randomness)
    {
        result.push(partial_sum.clone());
        partial_sum = partial_sum + power_of_randomness * (cur_limb.clone() - prev_limb.clone());
```

Suggestion because inside the iteration you have a particular power of randomness (in singular).